packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+
+    await E(lienBridge).setLiened(address, liened, updated);
+    // COMMIT. Burn the escrowed attestation payment and exit the seat
+    amountByAddress.set(address, updated);
+    zcfMint.burnLosses(harden({ Attestation: attestationAmount }), seat);
+    seat.exit();
+  };
+
+  /**
+   * Get liened amount synchronously.
+   *
+   * @param {string} address
+   * @param {Brand<'nat'>} brand
+   */
+  const getLiened = (address, brand) => {
+    assert.typeof(address, 'string');
+    assert.equal(
+      brand,
+      stakeBrand,
+      X`This contract can only make attestations for ${brand}`,
+    );
+    return getOrElse(amountByAddress, address, () => empty);
+  };
+
+  const lienMint = harden({
+    mintAttestation,
+    returnAttestation,
+    getLiened,
+    /** @param { Amount<'copyBag'> } attAmt */
+    unwrapLienedAmount: attAmt => unwrapLienedAmount(attAmt).lienedAmount,
+  });
+  return harden({ issuer, brand: attBrand, lienMint });
+};
+
+/**
+ * Authorize each account holder to lien some of their staked assets
+ * and get an attestation that the lien is in place.
+ *
+ * @param {ZCF} zcf
+ * @param {Brand} stakeBrand
+ * @param {string} keyword for attestation issuer
+ * @param {ERef<StakingAuthority>} lienBridge
+ *
+ * NOTE: the liened amount is kept both here in JS and on the
+ * other side of the lienBridge. The mint created here
+ * is the authority on the liened amount. The liened amount
+ * on the other side of the bridge may lag. Do not create two
+ * mints that share a bridge.
+ *
+ * The provideAttestationMaker method on the returned creatorFacet
+ * provides the authorization for an address. The account holder
+ * can then use attMaker.makeAttestation(lienDelta).
+ *
+ * Note the Attestation keyword for use in returnAttestation offers.
+ */
+export const makeAttestationFacets = async (
+  zcf,
+  stakeBrand,
+  keyword,
+  lienBridge,
+) => {
+  /** @type {Store<Address, AttMaker>} */
+  const attMakerByAddress = makeStore('address');
+
+  const { issuer, brand, lienMint } = await makeAttestationKit(
+    zcf,
+    keyword,
+    stakeBrand,
+    lienBridge,
+  );
+
+  /** @param { string } address */
+  const makeAttMaker = address =>
+    Far('attMaker', {
+      /** @param { Amount<'nat'> } lienedDelta */
+      makeAttestation: lienedDelta =>
+        lienMint.mintAttestation(address, lienedDelta),
+      getLiened: () => lienMint.getLiened(address, stakeBrand),
+      getAccountState: () => E(lienBridge).getAccountState(address, stakeBrand),
+      makeReturnAttInvitation: () =>
+        zcf.makeInvitation(lienMint.returnAttestation, 'returnAttestation'),
+      unwrapLienedAmount: lienMint.unwrapLienedAmount,
+    });
+
+  return harden({
+    publicFacet: Far('attestation publicFacet', {
+      getIssuer: () => issuer,
+      getBrand: () => brand,
+      makeReturnAttInvitation: () =>
+        zcf.makeInvitation(lienMint.returnAttestation, 'returnAttestation'),
+    }),
+    creatorFacet: Far('creator', {
```

Comment:
Well, they are facets of a (lien) attestation service; there's no particular information about being used in runStake here. They work just fine as their own contract; I only recently deleted the `start` function, mostly to avoid bundling time when testing.

I'm more inclined to clarify the separation as follows:

```js
  const creatorFacet = wrapCreatorFacet(
    Far('runStake creator', {
      provideAttestationMaker: att.creatorFacet.provideAttestationMaker,
      getLiened: att.creatorFacet.getLiened,
    }),
  );
```

Reply:
I like that.